week-1-ts-ecosystem/w01e04-light-me-up/src/task2/LightBulb.ts

```diff
@@ -0,0 +1,30 @@
+import { PowerSource } from "./PowerSource";
+
+export class LightBulb {
+  name: string;
+  protected readonly powerConsumption = 20;
+
+  constructor(name: string) {
+    this.name = name;
+  }
+
+  plugIn(powerSource: PowerSource, numberOfSeconds: number) {
+    if (numberOfSeconds < 1) {
+      throw new Error("Lightbulb needs to light at least 1 second.");
+    }
+    let counter = 0;
+    const lightUp = setInterval(() => {
+      if (numberOfSeconds === ++counter) {
+        clearInterval(lightUp);
+      }
+      try {
+        powerSource.consume(this.powerConsumption);
+        console.log(
+          `Lightbulb: ${this.name} consumes ${this.powerConsumption} W in a second.`
+        );
+      } catch (e) {
+        throw e;
```

Comment:
ten `throw` nie będzie miał tutaj znaczenia - zobaczysz to jak w `main.ts` spróbujesz zaimplementować jego obsługę